packages/mongodb-dialects/mongosh/src/main/kotlin/com/mongodb/jbplugin/dialects/mongosh/MongoshDialectFormatter.kt

```diff
@@ -1,16 +1,27 @@
 package com.mongodb.jbplugin.dialects.mongosh
 
 import com.mongodb.jbplugin.dialects.DialectFormatter
-import com.mongodb.jbplugin.mql.BsonType
-import com.mongodb.jbplugin.mql.IndexAnalyzer
-import com.mongodb.jbplugin.mql.Node
-import com.mongodb.jbplugin.mql.components.HasCollectionReference
-import com.mongodb.jbplugin.mql.components.HasTargetCluster
+import com.mongodb.jbplugin.dialects.mongosh.backend.MongoshBackend
+import com.mongodb.jbplugin.mql.*
+import com.mongodb.jbplugin.mql.components.*
 import io.github.z4kn4fein.semver.Version
 import org.owasp.encoder.Encode
 
 object MongoshDialectFormatter : DialectFormatter {
-    override fun <S> formatQuery(query: Node<S>, explain: Boolean) = ""
+    override fun <S> formatQuery(query: Node<S>, explain: Boolean): String = MongoshBackend().apply {
+        emitDbAccess()
+        emitCollectionReference(query.component<HasCollectionReference<S>>())
+        emitFunctionName("find")
+        emitFunctionCall({
+            emitQueryBody(query, firstCall = true)
+        })
+        if (explain) {
+            emitPropertyAccess()
+            emitFunctionName("explain")
+            emitFunctionCall()
+        }
+    }.computeOutput()
+
     override fun <S> indexCommandForQuery(query: Node<S>): String = when (val index = IndexAnalyzer.analyze(query)) {
```

Comment:
At some point we might want to improve the MongoshBackend to also generate the index creation script through the backend itself, instead of templating. However, it's not something really urgent at all so we can just do it whenever the index creation script gets more complicated.